src/lib/services/group-service.ts

```diff
@@ -170,8 +170,18 @@ export class GroupService {
         return [];
     }
 
-    async deleteGroup(id: number): Promise<void> {
-        return this.groupStore.delete(id);
+    async deleteGroup(id: number, userName?: string): Promise<void> {
+        const group = await this.groupStore.get(id);
+
+        await this.groupStore.delete(id);
+
+        if (userName) {
+            await this.eventStore.store({
+                type: GROUP_DELETED,
+                createdBy: userName,
+                data: group,
+            });
+        }
```

Comment:
This uses expand contract so we don't break Enterprise, that's why you see:
 - `userName?`
 - `if (userName)`

This will get cleaned up after we merge the Enterprise PR.